extensions/InputBox/InputBox.hooks.php

```diff
@@ -52,24 +52,38 @@ public static function onMediaWikiPerformAction(
 		$request, 
 		$wiki )
 	{
+		global $wgScript, $wgUser;
+
 		if( $wiki->getAction( $request ) !== 'edit' ){
 			# not our problem
 			return true;
 		}
+
+		$params = $request->getValues();
+
+		if ( $request->getText( 'preload', '' ) !== '' ){
+			# Make sure any call with preload uses VE when it matches the user's prefs
+			if ( EditorPreference::isVisualEditorPrimary() && $wgUser->isLoggedIn() ) {
+				$params['veaction'] = 'edit';
+				unset( $params['action'] );
+
+				$output->redirect( wfAppendQuery( $wgScript, $params ), '301' );
+				return false;
```

Comment:
I tested, and it looks to me like I could simplify the checks to all use the same format.

If my understanding is correct, checking for ```$request->getText( 'prefix', '' ) === ''``` is equivalent to checking for ```empty( $params['prefix'] )```. 

For reference, the url want to pick up generally looks like this:
http://visualeditor.pgroland.wikia-dev.com/index.php?action=edit&preload=MediaWiki%3ACharacter-preload&editintro=&summary=&nosummary=&prefix=&minor=&title=fdsfdsf3e423&create=Create+page

This sets prefix explicitly to an empty string, and it doesn't seem to matter if an empty parameter is present, or if it is not in the url at all, empty will still return true.